Harvest Billing Widget/widget.html

```diff
@@ -0,0 +1,201 @@
+            font-size: 1.0em;
+            margin-bottom: 0.3em;
+        }
+        .chart-container {
+            position: relative;
+            width: 100%;
+            height: 100%;
+        }
+    </style>
+</head>
+<body>
+    <div class="widget">
+        <h2 class="title">Billable/Non-billable Hours</h2>
+        <div class="chart-container">
+            <canvas id="hoursChart"></canvas>
+        </div>
+    </div>
+    <script type="text/javascript">
+        VSS.init({
+            explicitNotifyLoaded: true,
+            usePlatformStyles: true
+        });
+
+        VSS.require(["TFS/Dashboards/WidgetHelpers"], function (WidgetHelpers) {
+            WidgetHelpers.IncludeWidgetStyles();
+            VSS.register("HarvestHoursWidget", function () {
+                return {
+                    load: function (widgetSettings) {
+                        var settings = JSON.parse(widgetSettings.customSettings.data || "{}");
+                        var harvestAccountId = settings.harvestAccountId;
+                        var authToken = settings.authToken;
+                        var widgetTitle = widgetSettings.name || "Billable/Non-billable Hours";
+                        var displayMode = settings.displayMode || "hours";
+
+                        if (!harvestAccountId || !authToken) {
+                            console.error("Harvest Account ID or Authorization token is missing.");
+                            return WidgetHelpers.WidgetStatusHelper.Failure("Configuration is missing.");
+                        }
+
+                        // Set the widget title
+                        document.querySelector('.title').textContent = widgetTitle;
+
+                        var billableHours = 0;
+                        var nonBillableHours = 0;
+                        var rAndDHours = 0;
+
+                        var currentDate = new Date();
+                        var startDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), 1).toISOString().split('T')[0];
+                        var endDate = new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, 0).toISOString().split('T')[0];
+
+                        var harvestApiUrl = `https://api.harvestapp.com/v2/time_entries?from=${startDate}&to=${endDate}`;
+
+                        $.ajax({
+                            url: harvestApiUrl,
+                            method: 'GET',
+                            headers: {
+                                'Authorization': `Bearer ${authToken}`,
+                                'Harvest-Account-Id': harvestAccountId,
+                                'User-Agent': 'YourApp (yourname@example.com)'
+                            },
+                            success: function (data) {
+                                data.time_entries.forEach(function (entry) {
+                                    if (entry.project && entry.project.code && entry.project.code.includes("R&D")) {
```

Comment:
As discussed, please don't hard code the filter "R&D", this should be a widget property.

Reply:
Let me clarify, so we can show R&D or not depending on whether we choose it in the widget settings?

Reply:
A new R&D Filter property widget was created, allowing you to enter a word to filter and select projects.

![image](https://github.com/user-attachments/assets/7040ac75-efcf-4d54-b8b0-f2cb4abf070e)